XO.py

```diff
@@ -0,0 +1,157 @@
+        print()
+        print('***')
+        print(f'The winner is {players[curr_player].name}!!!')
+        print('***')
+
+    def declare_tie(self):
+        print()
+        print('***')
+        print('The result is a tie!!!')
+        print('***')
+
+
+class Controller:
+    def __init__(self, model, view):
+        self.model = model
+        self.view = view
+        self.players, self.curr_player = self.initiate_players()
+        self.number_of_moves_left = N * N  # Equals to number of squares on board.
+
+    def initiate_players(self):
+        Player = namedtuple('Player', 'name symbol')
+        x_player_name = input('Enter X player name\n')
+        o_player_name = input('Enter O player name\n')
+        while x_player_name == o_player_name:
+            o_player_name = input('Both names are the same. Choose a different name for O player\n')
+        players = [Player(name=x_player_name, symbol='x'), Player(name=o_player_name, symbol='o')]
+        curr_player = randint(0, 1)
+        return players, curr_player
+
+    def show_board(self):
+        board = self.model.get_board_status()
+        self.view.print_board(board)
+
+    def get_input(self):
+        line = self.is_input_valid('line')
+        col = self.is_input_valid('column')
+        updated = self.model.update_board(line, col, self.players, self.curr_player)
+        if not updated:
+            print(f'{self.players[self.curr_player].name}, you entered a taken spot on the board, try again')
+            return False
+        return True
+
+    def is_input_valid(self, input_type):
+        valid_input = False
+        while not valid_input:
+            print(f'{self.players[self.curr_player].name}, please insert the {input_type} number')
+            try:
+                input_value = int(input())
+            except ValueError:
+                print(f'{self.players[self.curr_player].name}, input not valid')
+                continue
+            if input_value < 0 or input_value >= N:
+                print(f'{self.players[self.curr_player].name}, insert a {input_type} num between 0 to {N-1}, including')
+            else:
+                valid_input = True
+        return input_value
+
+    def switch_player(self):
+        self.curr_player = self.curr_player ^ 1
+
+    def is_win(self):
+        if self.model.line_won(self.players, self.curr_player) \
+                or self.model.col_won(self.players, self.curr_player) \
+                or self.model.diagonal_won(self.players, self.curr_player):
+            self.view.declare_winner(self.players, self.curr_player)
+            self.show_board()
+            return True
+        return False
+
+    def is_tie(self):
+        self.number_of_moves_left -= 1
+        if self.number_of_moves_left == 0:
+            self.view.declare_tie()
+            self.show_board()
+            return True
+        return False
+
+    def main(self):
+        game_won = False
+        game_tie = False

```

Comment:
Why not just use a single variable - `game_done`?

Reply:
Done